packages/common/src/content.ts

```diff
@@ -885,6 +891,66 @@ const getSolutionUrl = (content: IHubContent): string => {
   return hubUrl;
 };
 
+/**
+ * Create a new content with updated boundary properties
+ * @param content original content
+ * @param boundary boundary provenance
+ * @returns
+ */
+export const setContentBoundary = (
+  content: IHubContent,
+  boundary: GeographyProvenance
+) => {
+  // update content's item and boundary
+  const properties = { ...(content.item.properties || {}), boundary };
+  const item = { ...content.item, properties };
+  const updated = { ...content, item };
+  return { ...updated, boundary: getContentBoundary(updated) };
+};
+
+/**
+ * Create a new content with updated extent and derived properties like boundary
+ * @param content original content
+ * @param extent new extent
+ * @returns a new content with the updated extent and boundary
+ */
+export const setContentExtent = (
+  content: IHubContent,
+  extent: IBBox
+): IHubContent => {
+  // update content's item and extent
+  const item = { ...content.item, extent };
+  const updated = { ...content, item, extent };
```

Comment:
Does this mean that `extent` becomes a property of both `content` and `content.item`?

Reply:
Unfortunately, all `content.item` properties are mirrored on `content` for backwards compatibility reasons, though in retrospect I wish I had chosen not to do that when we moved to Hub.js, even though it would have meant updating hundreds of LOC in the -ui app.